watertap/unit_models/tests/test_uv_aop.py

```diff
@@ -207,32 +221,539 @@ def test_solve(self, UV_frame):
+
+    @pytest.mark.component
+    def test_initialize(self, UV_frame):
+        initialization_tester(UV_frame)
+
+    @pytest.mark.component
+    def test_var_scaling(self, UV_frame):
+        m = UV_frame
+        badly_scaled_var_lst = list(badly_scaled_var_generator(m))
+        [print(i[0].name, i[1]) for i in badly_scaled_var_lst]
+        assert badly_scaled_var_lst == []
+
+    @pytest.mark.component
+    def test_solve(self, UV_frame):
+        m = UV_frame
+        results = solver.solve(m)
+
+        # Check for optimal solution
+        assert results.solver.termination_condition == TerminationCondition.optimal
+        assert results.solver.status == SolverStatus.ok
+
+    @pytest.mark.component
+    def test_solution(self, UV_frame):
+        m = UV_frame
+        assert pytest.approx(1206.5, rel=1e-3) == value(
+            m.fs.unit.control_volume.properties_in[0].flow_mass_phase_comp["Liq", "H2O"]
+        )
+        assert pytest.approx(4356.4694, rel=1e-3) == value(
+            pyunits.convert(
+                m.fs.unit.control_volume.properties_in[0].flow_vol,
+                to_units=pyunits.m**3 / pyunits.hr,
+            )
+        )
+        assert pytest.approx(8.9281e-05, rel=1e-3) == value(
+            m.fs.unit.control_volume.properties_in[0].flow_mass_phase_comp[
+                "Liq", "NDMA"
+            ]
+        )
+        assert pytest.approx(32, rel=1e-3) == value(
+            pyunits.convert(m.fs.unit.uv_dose, to_units=pyunits.mJ / pyunits.cm**2)
+        )
+        assert pytest.approx(2.5, rel=1e-3) == value(
+            log10(
+                m.fs.unit.control_volume.properties_in[0].flow_mass_phase_comp[
+                    "Liq", "NDMA"
+                ]
+                / m.fs.unit.control_volume.properties_out[0].flow_mass_phase_comp[
+                    "Liq", "NDMA"
+                ]
+            )
+        )
+        assert pytest.approx(38.7242, rel=1e-3) == value(
+            pyunits.convert(m.fs.unit.reactor_volume, to_units=pyunits.m**3)
+        )
+        assert pytest.approx(240, rel=1e-3) == value(
+            pyunits.convert(m.fs.unit.electricity_demand[0], to_units=pyunits.kW)
         )
-        assert pytest.approx(1829.525, rel=1e-3) == value(
-            m.fs.unit.electricity_demand_phase_comp[0, "Liq", "NDMA"]
+
+    @pytest.mark.requires_idaes_solver
+    @pytest.mark.component
+    def test_costing(self, UV_frame):
+        m = UV_frame
+
+        m.fs.costing = WaterTAPCosting()
+        m.fs.costing.base_currency = pyunits.USD_2020
+
+        m.fs.unit.costing = UnitModelCostingBlock(
+            default={
+                "flowsheet_costing_block": m.fs.costing,
+            },
+        )
+        m.fs.costing.cost_process()
+        m.fs.costing.add_LCOW(m.fs.unit.control_volume.properties_out[0].flow_vol)
+        results = solver.solve(m)
+
+        assert results.solver.termination_condition == TerminationCondition.optimal
```

Comment:
same as above- basically, replace with assert_optimal_termination or add assertion to check that solver status is OK